PureBasicIDE/ColorSchemes.pb

```diff
@@ -0,0 +1,324 @@
+﻿; --------------------------------------------------------------------------------------------
+;  Copyright (c) Fantaisie Software. All rights reserved.
+;  Dual licensed under the GPL and Fantaisie Software licenses.
+;  See LICENSE and LICENSE-FANTAISIE in the project root for license information.
+; --------------------------------------------------------------------------------------------
+
+
+; Store color names in an array, for indexable lookup
+Global Dim ColorName.s(#COLOR_Last_IncludingToolsPanel)
+
+; Color Scheme structure
+Structure ColorSchemeStruct
+  Name.s
+  File.s
+  
+  ColorValue.l[#COLOR_Last_IncludingToolsPanel + 1]
+  
+  IsIDEDefault.i
+  IsAccessibility.i
+EndStructure
+
+; Global list of built-in / found schemes
+Global NewList ColorScheme.ColorSchemeStruct()
+
+; Color values with special meanings
+#ColorSchemeValue_UseSysColor = -1
+#ColorSchemeValue_Undefined   = -2
+
+
+; These (from Preferences.pb) are used
+Global PreferenceToolsPanelFrontColor, PreferenceToolsPanelBackColor
+Declare UpdatePreferenceSyntaxColor(ColorIndex, Color)
+Declare UpdateImageColorGadget(Gadget, Image, Color)
+
+
+; Returns #True if the specified color scheme matches the user's current color settings, otherwise #False
+Procedure.i ColorSchemeMatchesCurrentSettings(*ColorScheme.ColorSchemeStruct)
+  Protected Result.i = #True
+  
+  For i = 0 To #COLOR_Last
+    If ((i <> #COLOR_Selection) And (i <> #COLOR_SelectionFront)) ; selection colors may follow OS, so skip them for scheme match check
```

Comment:
Please remove unneeded braces. Only add them when there are multiple And/Or statement to avoid ambiguity. Especially when there is only one statement, adding braces only reduces readability.

Reply:
Will do. In this highlighted example, would you remove just the outer parentheses, or all the parentheses?

Reply:
I'd remove all of them. Only use parentheses when they make a difference. The only exception to this is when And/Or are mixed in a statement because PB has a bit weird precedence rules here, so I tend to add them in this case just so it is clear to everybody what is happening.